File: appfw/src/main/java/com/jx/appfw/data/repository/datasource/impl/MainDataStoreNetImpl.java
package com.jx.appfw.data.repository.datasource.impl;

import com.jx.appfw.data.db.MainDB;
import com.jx.appfw.data.net.MainApi;
import com.jx.appfw.data.repository.datasource.MainDataStore;
import com.jx.appfw.domain.request.main.TestBean;

import io.reactivex.Observable;

/**
 * {@link MainDataStore} implementation for retrieving data from the network.
 */
public class MainDataStoreNetImpl implements MainDataStore {
    private final MainApi mainApi;

    private final MainDB mainDB;

    public MainDataStoreNetImpl(MainApi mainApi, MainDB mainDB) {
        this.mainApi = mainApi;
        this.mainDB = mainDB;
    }

    @Override
    public Observable<String> test(TestBean testBean) {
        return mainApi.test(testBean);
    }
}

File: appfw/src/main/java/com/jx/appfw/data/repository/datasource/impl/MainDataStoreNetImpl.kt
package com.jx.appfw.data.repository.datasource.impl

import com.jx.appfw.data.db.MainDB
import com.jx.appfw.data.net.MainApi
import com.jx.appfw.data.repository.datasource.MainDataStore
import com.jx.appfw.domain.request.main.TestBean
import io.reactivex.Observable

class MainDataStoreNetImpl(val mMainApi: MainApi, val mMainDB: MainDB) : MainDataStore {
    override fun test(testBean: TestBean): Observable<String> {
        return mMainApi.test(testBean)
    }
}